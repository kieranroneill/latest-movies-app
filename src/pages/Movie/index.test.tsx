import {
    shallow,
    ShallowWrapper
} from 'enzyme';
import * as React from 'react';

// Components.
import {
    Movie,
    Props
} from './';

// Mocks.
import { MockRouteComponentProps } from '../../../test/mocks/reactRouterMock';

interface Scope {
    props: Props;
    wrapper: ShallowWrapper;
}

describe('src/pages/Movie', () => {
    let scope: Scope;

    beforeEach(() => {
        const props: Props = {
            extendedMovie: {
                loading: false,
                result: {},
            },
            getExtendedMovie: jest.fn(),
            resetExtendedMovie: jest.fn(),
            setPageTitle: jest.fn(),
        };

        scope = {
            props,
            wrapper: shallow(<Movie {...props} {...new MockRouteComponentProps()} />),
        }
    });

    describe('<Movie /> snapshots', () => {
        it('should match the snapshot with default props', () => {
            expect(scope.wrapper).toMatchSnapshot();
        });
    });

    describe('when the component mounts', () => {
        it('should send a request to get the extended movie details', () => {
            expect(scope.props.getExtendedMovie).toBeCalled();
        });
    });

    describe('when the component unmounts', () => {
        it('should reset the extended movie details', () => {
            scope.wrapper.unmount();

            expect(scope.props.resetExtendedMovie).toBeCalled();
        });
    });
});
