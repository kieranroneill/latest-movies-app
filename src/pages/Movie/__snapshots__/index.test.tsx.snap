// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/Movie <Movie /> snapshots should match the snapshot with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    extendedMovie={
      Object {
        "loading": false,
        "result": Object {},
      }
    }
    getExtendedMovie={
      [MockFunction] {
        "calls": Array [
          Array [
            NaN,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    history={
      MockHistory {
        "action": "POP",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 2,
        "listen": [MockFunction],
        "location": MockLocation {
          "hash": "",
          "key": "",
          "pathname": "/",
          "search": "",
          "state": Object {},
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
    location={
      MockLocation {
        "hash": "",
        "key": "",
        "pathname": "/",
        "search": "",
        "state": Object {},
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
    resetExtendedMovie={[MockFunction]}
    setPageTitle={[MockFunction]}
    staticContext={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <styled.div>
        <styled.div />
        <styled.div>
          <Link
            replace={false}
            to="/movies"
          >
            <Styled(WithStyles(Icon))
              style={
                Object {
                  "fontSize": "6rem",
                }
              }
            >
              keyboard_arrow_left
            </Styled(WithStyles(Icon))>
          </Link>
          <styled.div>
            <styled.div>
              <styled.h1>
                
              </styled.h1>
              <styled.h2>
                0
              </styled.h2>
              <Styled(WithStyles(Icon))
                style={
                  Object {
                    "fontSize": "4rem",
                  }
                }
              >
                grade
              </Styled(WithStyles(Icon))>
            </styled.div>
            <styled.div />
            <styled.h4>
              
            </styled.h4>
          </styled.div>
        </styled.div>
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div />,
          <styled.div>
            <Link
              replace={false}
              to="/movies"
            >
              <Styled(WithStyles(Icon))
                style={
                  Object {
                    "fontSize": "6rem",
                  }
                }
              >
                keyboard_arrow_left
              </Styled(WithStyles(Icon))>
            </Link>
            <styled.div>
              <styled.div>
                <styled.h1>
                  
                </styled.h1>
                <styled.h2>
                  0
                </styled.h2>
                <Styled(WithStyles(Icon))
                  style={
                    Object {
                      "fontSize": "4rem",
                    }
                  }
                >
                  grade
                </Styled(WithStyles(Icon))>
              </styled.div>
              <styled.div />
              <styled.h4>
                
              </styled.h4>
            </styled.div>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageUrl": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/movies"
              >
                <Styled(WithStyles(Icon))
                  style={
                    Object {
                      "fontSize": "6rem",
                    }
                  }
                >
                  keyboard_arrow_left
                </Styled(WithStyles(Icon))>
              </Link>,
              <styled.div>
                <styled.div>
                  <styled.h1>
                    
                  </styled.h1>
                  <styled.h2>
                    0
                  </styled.h2>
                  <Styled(WithStyles(Icon))
                    style={
                      Object {
                        "fontSize": "4rem",
                      }
                    }
                  >
                    grade
                  </Styled(WithStyles(Icon))>
                </styled.div>
                <styled.div />
                <styled.h4>
                  
                </styled.h4>
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(WithStyles(Icon))
                  style={
                    Object {
                      "fontSize": "6rem",
                    }
                  }
                >
                  keyboard_arrow_left
                </Styled(WithStyles(Icon))>,
                "replace": false,
                "to": "/movies",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "keyboard_arrow_left",
                  "style": Object {
                    "fontSize": "6rem",
                  },
                },
                "ref": null,
                "rendered": "keyboard_arrow_left",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div>
                    <styled.h1>
                      
                    </styled.h1>
                    <styled.h2>
                      0
                    </styled.h2>
                    <Styled(WithStyles(Icon))
                      style={
                        Object {
                          "fontSize": "4rem",
                        }
                      }
                    >
                      grade
                    </Styled(WithStyles(Icon))>
                  </styled.div>,
                  <styled.div />,
                  <styled.h4>
                    
                  </styled.h4>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.h1>
                        
                      </styled.h1>,
                      <styled.h2>
                        0
                      </styled.h2>,
                      <Styled(WithStyles(Icon))
                        style={
                          Object {
                            "fontSize": "4rem",
                          }
                        }
                      >
                        grade
                      </Styled(WithStyles(Icon))>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": 0,
                      },
                      "ref": null,
                      "rendered": 0,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "grade",
                        "style": Object {
                          "fontSize": "4rem",
                        },
                      },
                      "ref": null,
                      "rendered": "grade",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <styled.div>
          <styled.div />
          <styled.div>
            <Link
              replace={false}
              to="/movies"
            >
              <Styled(WithStyles(Icon))
                style={
                  Object {
                    "fontSize": "6rem",
                  }
                }
              >
                keyboard_arrow_left
              </Styled(WithStyles(Icon))>
            </Link>
            <styled.div>
              <styled.div>
                <styled.h1>
                  
                </styled.h1>
                <styled.h2>
                  0
                </styled.h2>
                <Styled(WithStyles(Icon))
                  style={
                    Object {
                      "fontSize": "4rem",
                    }
                  }
                >
                  grade
                </Styled(WithStyles(Icon))>
              </styled.div>
              <styled.div />
              <styled.h4>
                
              </styled.h4>
            </styled.div>
          </styled.div>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div />,
            <styled.div>
              <Link
                replace={false}
                to="/movies"
              >
                <Styled(WithStyles(Icon))
                  style={
                    Object {
                      "fontSize": "6rem",
                    }
                  }
                >
                  keyboard_arrow_left
                </Styled(WithStyles(Icon))>
              </Link>
              <styled.div>
                <styled.div>
                  <styled.h1>
                    
                  </styled.h1>
                  <styled.h2>
                    0
                  </styled.h2>
                  <Styled(WithStyles(Icon))
                    style={
                      Object {
                        "fontSize": "4rem",
                      }
                    }
                  >
                    grade
                  </Styled(WithStyles(Icon))>
                </styled.div>
                <styled.div />
                <styled.h4>
                  
                </styled.h4>
              </styled.div>
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "imageUrl": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/movies"
                >
                  <Styled(WithStyles(Icon))
                    style={
                      Object {
                        "fontSize": "6rem",
                      }
                    }
                  >
                    keyboard_arrow_left
                  </Styled(WithStyles(Icon))>
                </Link>,
                <styled.div>
                  <styled.div>
                    <styled.h1>
                      
                    </styled.h1>
                    <styled.h2>
                      0
                    </styled.h2>
                    <Styled(WithStyles(Icon))
                      style={
                        Object {
                          "fontSize": "4rem",
                        }
                      }
                    >
                      grade
                    </Styled(WithStyles(Icon))>
                  </styled.div>
                  <styled.div />
                  <styled.h4>
                    
                  </styled.h4>
                </styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(WithStyles(Icon))
                    style={
                      Object {
                        "fontSize": "6rem",
                      }
                    }
                  >
                    keyboard_arrow_left
                  </Styled(WithStyles(Icon))>,
                  "replace": false,
                  "to": "/movies",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "keyboard_arrow_left",
                    "style": Object {
                      "fontSize": "6rem",
                    },
                  },
                  "ref": null,
                  "rendered": "keyboard_arrow_left",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div>
                      <styled.h1>
                        
                      </styled.h1>
                      <styled.h2>
                        0
                      </styled.h2>
                      <Styled(WithStyles(Icon))
                        style={
                          Object {
                            "fontSize": "4rem",
                          }
                        }
                      >
                        grade
                      </Styled(WithStyles(Icon))>
                    </styled.div>,
                    <styled.div />,
                    <styled.h4>
                      
                    </styled.h4>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <styled.h1>
                          
                        </styled.h1>,
                        <styled.h2>
                          0
                        </styled.h2>,
                        <Styled(WithStyles(Icon))
                          style={
                            Object {
                              "fontSize": "4rem",
                            }
                          }
                        >
                          grade
                        </Styled(WithStyles(Icon))>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "",
                        },
                        "ref": null,
                        "rendered": "",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": 0,
                        },
                        "ref": null,
                        "rendered": 0,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "grade",
                          "style": Object {
                            "fontSize": "4rem",
                          },
                        },
                        "ref": null,
                        "rendered": "grade",
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
